### PR TITLE
fix: change the size limit of sql stmt rebuilt for show create table to TSDB_MAX_SQL_LEN

### PR DESCRIPTION
**Pull request by bgzhao66**

### CODE REVIEW

File: cmake/define.inc [hunk 1]
```diff
@@ -224,9 +224,9 @@ IF (TD_DARWIN_64)
   ADD_DEFINITIONS(-DUSE_LIBICONV)
   MESSAGE(STATUS "darwin64 is defined")
   IF ("${CPUTYPE}" STREQUAL "apple_m1")
-    SET(COMMON_FLAGS "-Wall -Werror -Wno-missing-braces -fPIC -D_FILE_OFFSET_BITS=64 -D_LARGE_FILE")
+    SET(COMMON_FLAGS "-Wall -Werror -Wno-missing-braces -Wno-tautological-constant-out-of-range-compare -Wno-unused-but-set-variable -fPIC -D_FILE_OFFSET_BITS=64 -D_LARGE_FILE")
```
Comment (bgzhao66) [inline cmake/define.inc:227]: To be consistent on compiler warnings among Darwin, Linux, ARM64 and Win OSes at the this point. 
In OSes other than Darwin, the compilers (i.e. GCC in Linux and so) ignores the tautological assert statements which are always true without complaining.

Comment (bgzhao66) [reply cmake/define.inc:227]: In Darwin, Clang's compiler warning on unused-but-set-variable is not accurate. Disable it too.

File: Jenkinsfile2 [hunk 2]
```diff
@@ -240,12 +240,12 @@ def pre_test_win(){
     }
     bat '''
         cd %WIN_INTERNAL_ROOT%
-        git pull
+        git pull  origin  ''' + env.CHANGE_TARGET + '''
```
Comment (bgzhao66) [inline Jenkinsfile2:243]: It is to avoid the git pull issue during CI, submitted by Haoran.

File: src/client/src/tscLocal.c [hunk 3]
```diff
@@ -295,7 +295,7 @@ void tscSCreateCallBack(void *param, TAOS_RES *tres, int code) {
     taos_fetch_rows_a(tres, tscSCreateCallBack, param);    
     builder->callStage = SCREATE_CALLBACK_RETRIEVE;
   } else {
-    char *result = calloc(1, TSDB_MAX_BINARY_LEN);
+    char *result = calloc(1, TSDB_MAX_SQL_LEN);
```
Comment (bgzhao66) [inline src/client/src/tscLocal.c:298]: Use TSDB_MAX_SQL_LEN to control the length of rebuilt sql stmt, instead of TSDB_MAX_BINARY_LEN.

**Review by dapan1121 — APPROVED** [review]

**Pull request merged**